pkg/log/format/options/color.go

```diff
@@ -0,0 +1,240 @@
+		val.value = ColorValue(byte(num))
+
+		return nil
+	}
+
+	if err := val.MapValue.Parse(str); err != nil {
+		return errors.Errorf("available values: 0..255,%s", strings.Join(maps.Values(val.list), ","))
+	}
+
+	return nil
+}
+
+type ColorScheme map[ColorValue]ColorStyle
+
+func (scheme ColorScheme) Compile() compiledColorScheme {
+	compiled := make(compiledColorScheme, len(scheme))
+
+	for name, val := range scheme {
+		compiled[name] = val.ColorFunc()
+	}
+
+	for i := range 255 {
+		s := strconv.Itoa(i)
+
+		compiled[ColorValue(i)] = ColorStyle(s).ColorFunc()
+	}
+
+	return compiled
+}
+
+type ColorStyle string
+
+func (val ColorStyle) ColorFunc() ColorFunc {
+	return ansi.ColorFunc(string(val))
+}
+
+type ColorFunc func(string) string
+
+type ColorValue int
+
+type compiledColorScheme map[ColorValue]ColorFunc
+
+type ColorOption struct {
+	*CommonOption[ColorValue]
+	compiledColors compiledColorScheme
+	gradientColor  *gradientColor
+}
+
+// Format implements `Option` interface.
+func (color *ColorOption) Format(data *Data, str string) (string, error) {
+	value := color.value.Get()
+
+	if value == NoneColor {
+		return str, nil
+	}
+
+	if value == DisableColor || data.DisableColors {
+		return log.RemoveAllASCISeq(str), nil
+	}
+
+	if value == PresetColor && data.PresetColorFn != nil {
+		value = data.PresetColorFn()
+	}
+
+	if value == GradientColor && color.gradientColor != nil {
+		value = color.gradientColor.Value(str)
+	}
+
+	if colorFn, ok := color.compiledColors[value]; ok {
+		str = colorFn(str)
+	}
+
+	return str, nil
+}
+
+// Color creates the option to change the color of text.
+func Color(val ColorValue) Option {
+	return &ColorOption{
+		CommonOption:   NewCommonOption(ColorOptionName, colorList.Set(val)),
+		compiledColors: colorScheme.Compile(),
+		gradientColor:  newGradientColor(),
+	}
+}
+
+var (
+	// defaultAutoColorValues contains ANSI color codes that are assigned sequentially to each unique text in a rotating order
+	// https://user-images.githubusercontent.com/995050/47952855-ecb12480-df75-11e8-89d4-ac26c50e80b9.png
+	// https://www.hackitu.de/termcolor256/
+	defaultAutoColorValues = []ColorValue{ //nolint:gochecknoglobals
+		66,
+		67,
+		95,
+		96,
+		102,
+		103,
+		108,
+		109,
+		138,
+		139,
+		144,
+		145,
+	}
+)
+
+type gradientColor struct {
+	// cache stores unique text with their color code.
+	// We use [xsync.MapOf](https://github.com/puzpuzpuz/xsync?tab=readme-ov-file#map) instead of standard `sync.Map` since it's faster and has generic types.
+	cache  *xsync.MapOf[string, ColorValue]
+	values []ColorValue
+
+	// nextStyleIndex is used to get the next style from the `codes` list for a newly discovered text.
+	nextStyleIndex int
+}
+
+func newGradientColor() *gradientColor {
+	return &gradientColor{
+		cache:  xsync.NewMapOf[string, ColorValue](),
+		values: defaultAutoColorValues,
+	}
+}
+
+func (color *gradientColor) Value(text string) ColorValue {
```

Comment:
We already discussed this in a call conversation. This can works as gradient, we need just to fill the [slice](https://github.com/gruntwork-io/terragrunt/blob/1a9dafde59bb97aa9a58853009d9df0eed1169f0/pkg/log/format/options/color.go#L191-L203) with gradient colors,